core/auth/src/main/java/software/amazon/awssdk/auth/credentials/ContainerCredentialsProvider.java

```diff
@@ -199,19 +215,45 @@ public ResourcesEndpointRetryPolicy retryPolicy() {
         public Map<String, String> headers() {
             Map<String, String> requestHeaders = new HashMap<>();
             requestHeaders.put("User-Agent", SdkUserAgent.create().userAgent());
-            SdkSystemSetting.AWS_CONTAINER_AUTHORIZATION_TOKEN.getStringValue()
+            getTokenValue()
                                                               .filter(StringUtils::isNotBlank)
                                                               .ifPresent(t -> requestHeaders.put("Authorization", t));
             return requestHeaders;
         }
 
+        private Optional<String> getTokenValue() {
+            if (SdkSystemSetting
+                .AWS_CONTAINER_AUTHORIZATION_TOKEN.getStringValue().isPresent()) {
+                return SdkSystemSetting
+                    .AWS_CONTAINER_AUTHORIZATION_TOKEN
+                    .getStringValue();
+            }
+
+
+            return SdkSystemSetting.AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE.getStringValue()
+                                                                          .map(this::readToken);
+
+        }
+
+        private String readToken(String filePath) {
+            Path path = Paths.get(filePath);
+            if (!Files.exists(path)) {
+                throw SdkClientException.create(
+                    String.format("Failed to read authorization token from '%s': no such file or directory", filePath));
+            }
```

Comment:
I don't think this check is necessary; it's also not atomic since the file could exist during this check but not exist when we try to run it. Can we just to `Files.readAllBytes` below and catch the exception for file not exists?

Reply:
The Specs mentioned 
```
      "reason": "failed to read authorization token from '/full/path/to/token/file': no such file or directory"
  ```
  
  Thus I had added this specific case.

But , I think we can let user use the sdkException.getCause() which will give NoSuchFileException. Updated the code

Reply:
Gotcha. I think without this change, we will still meet that requirement